src/app/modules/payments/stripe.controller.ts

```diff
@@ -361,28 +361,36 @@ export const downloadPaymentInvoice: ControllerHandler<{
           // convert to pdf and return
           .then((receiptUrlResponse) => {
```

Comment:
Maybe for another ticket/PR,

but was wondering if we might want to abstract out the entire section for converting the receipt to url (line 363-400) conversion to `payment.service.ts`

this may help us to write unit tests to ensure the behaviour is correct e.g.
- will use the form address/gst before agency if available
- will still generate invoice without address/gst (which may be in the future this might be an error?)

It may also be a bit cleaner on our controller `.then((receiptUrlResponse) => convertToInvoicePdf(receiptUrlResponse, populatedForm))`

Reply:
Yeah makes sense!